store/store.go

```diff
@@ -379,23 +385,55 @@ func (s *Store[H]) prepareDeleteRangeBatch(
 	return nil
 }
 
-func (s *Store[H]) updateTail(
-	ctx context.Context, batch datastore.Batch, to uint64,
-) (H, error) {
-	var zero H
+func (s *Store[H]) updateTail(ctx context.Context, batch datastore.Batch, to uint64) error {
+	head, err := s.Head(ctx)
+	if err != nil {
+		return err
+	}
 
 	newTail, err := s.getByHeight(ctx, to)
 	if err != nil {
 		if !errors.Is(err, header.ErrNotFound) {
-			return zero, fmt.Errorf("cannot fetch next tail: %w", err)
+			return fmt.Errorf("cannot fetch next tail: %w", err)
 		}
-		return zero, err
+
+		if to <= head.Height() {
```

Comment:
I would add comment here for what err we're trying to catch -- or wrap err.

My understanding from sync discussion was that we're just doing this check here to ensure that we don't de-init the store if we didn't delete everything (where `to` is still below or == to `head` as `to` is non-inclusive)

Reply:
Yes, that's right

Reply:
> I would add comment here for what err we're trying to catch -- or wrap err.

There error there is always going to ErrNotFound, because all others are caught earlier. The idea there is to return this error saying that what user wants to delete actually doesn't exist.

Reply:
added an explicit error there